components/postmark/postmark.app.mjs

```diff
@@ -1,11 +1,208 @@
+    metadata: {
+      type: "object",
+      label: "Metadata",
+      description: "Custom metadata key/value pairs.",
+      optional: true,
+    },
+    message_stream: {
+      type: "string",
+      label: "Message stream",
+      description:
+        "Set message stream ID that's used for sending. If not provided, message will default to the outbound transactional stream.",
+      optional: true,
+    },
+  },
   methods: {
     // this.$auth contains connected account data
     authKeys() {
       console.log(Object.keys(this.$auth));
     },
+    async listTemplates() {
+      return await axios({
+        url: "https://api.postmarkapp.com/templates?count=500&offset=0",
+        headers: this.getHeaders(),
+        method: "GET",
+      })
+        .then(({ data }) =>
+          data.TotalCount
+            ? data.Templates.filter(
+              (obj) => obj.TemplateType === "Standard",
+            ).map((obj) => {
+              return {
+                label: obj.Name,
+                value: obj.Alias,
+              };
+            })
+            : [
+              {
+                label:
+                    "No templates found for this Postmark account. Create a template, then refresh this field.",
+                value: -1,
+              },
+            ])
+        .catch(() => [
+          {
+            label:
+              "An error ocurred while fetching the list of templates for this Postmark account. Please try again.",
+            value: -2,
+          },
+        ]);
+    },
+    getHeaders() {
+      return {
+        "X-Postmark-Server-Token": this.$auth.api_key,
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+      };
+    },
+    listSharedProps() {
+      return [
+        "from_email",
+        "to_email",
+        "cc_email",
+        "bcc_email",
+        "tag",
+        "reply_to",
+        "custom_headers",
+        "track_opens",
+        "track_links",
+        "attachments",
+        "metadata",
+        "message_stream",
+      ];
+    },
+    async sharedActionRequest($, action, endpoint, uniqueProps) {
+      return await axiosPipedream($, {
+        url: `https://api.postmarkapp.com/${endpoint}`,
+        headers: this.getHeaders(),
+        method: "POST",
+        data: {
+          ...uniqueProps,
+          From: action.from_email,
+          To: action.to_email,
+          Cc: action.cc_email,
+          Bcc: action.bcc_email,
+          Tag: action.tag,
+          ReplyTo: action.reply_to,
+          Headers: action.custom_headers,
+          TrackOpens: action.track_opens,
+          TrackLinks: action.track_links,
+          Attachments: action.attachments?.map((str) => {
+            let params = str.split("|");
+            return params.length === 3
+              ? {
+                Name: params[0],
+                Content: params[1],
+                ContentType: params[2],
+              }
+              : JSON.parse(str);
+          }),
```

Comment:
Could this be in a separate method?

Actually, could `Name`, `Content` and `ContentType` be separate props and `Attachments` object be constructed inside `async run()`?

Reply:
Moving it to a separate method, certainly agree it keeps things more organized.

The problem with making the props separate is it does not feel intuitive or user-friendly at all, in my opinion. Namely because there can be any number of attachments, each of which need these three props. Given an example of three attachments, in the current implementation:

```
Attachments: [
  "{ Name: 'readme.txt', Content: att1content, ContentType: 'text/plain' }",
  "{ Name: 'photo.png', Content: att2content, ContentType: 'image/png' }",
  "{ Name: 'index.html', Content: att3content, ContentType: 'text/html' }" ]
```

And by making them separate props, the individual data is in a more straightforward format, but the attachments are "split into pieces":

Attachment Names: [
  "readme.txt",
  "photo.png",
  "index.html" ]
Attachment Contents: [
  att1content,
  att2content,
  att3content ]
Attachment ContentTypes: [
  "text/plain",
  "image/png",
  "text/html" ]

I could not think of a better way to implement this, but any ideas are welcome, as I do not think the current implementation is optimal either.

Reply:
Yeah, I see your point. We don't have a great way to do this yet, so generally we try to keep the number of options low, but in this case I think it's fine for now